### PR TITLE
[PIO-57] Add SBT Native Packager

### PR DESCRIPTION
**Pull request by marevol**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ env:
     - BUILD_TYPE=Unit METADATA_REP=PGSQL EVENTDATA_REP=PGSQL MODELDATA_REP=PGSQL
     - BUILD_TYPE=Integration METADATA_REP=PGSQL EVENTDATA_REP=PGSQL MODELDATA_REP=PGSQL
     - BUILD_TYPE=Integration METADATA_REP=ELASTICSEARCH EVENTDATA_REP=HBASE MODELDATA_REP=LOCALFS ES_VERSION=5
-    - BUILD_TYPE=Integration METADATA_REP=ELASTICSEARCH EVENTDATA_REP=PGSQL MODELDATA_REP=HDFS ES_VERSION=5
+    - BUILD_TYPE=Integration METADATA_REP=ELASTICSEARCH EVENTDATA_REP=ELASTICSEARCH MODELDATA_REP=HDFS ES_VERSION=5
```
Comment (dszeto) [inline .travis.yml]: Let's create a new entry if you want to test a new combination.

Comment (marevol) [inline .travis.yml]: Okay, I'll create a new one.

**Comment by marevol** [comment]
>  I am wondering about the reason behind moving bin and conf to assembly

SBT native packager is contained in assembly directory, and putting them to src/universal is conventions of  sbt packager.
I'll check if we keep bin and conf in a project root directory.

**Comment by dszeto** [comment]
This is nice work! Would it be easier to wrap it with a script like what we are doing with make-distribution.sh? Also we are looking to release 0.11 soon once PIO-30 finished testing, so let me know if you need help to get this in, or set it aside for the next release.

Also hoping to put up a release roadmap after 0.11 for discussion.

**Comment by marevol** [comment]
Using make-distribution.sh, I'll try to modify it to keep the directory structure.

**Comment by marevol** [comment]
bin and conf are back, and I also added --with-rpm/deb option to make-distribution.sh.

**Comment by marevol** [comment]
Synced up with develop branch.

**Comment by dszeto** [comment]
Hey @marevol , running into this:

```
+ sbt/sbt assembly/rpm:packageBin
[info] Loading global plugins from /Users/dszeto/.sbt/0.13/plugins
[info] Loading project definition from /Users/dszeto/asf/incubator-predictionio/project
[info] Set current project to apache-predictionio-parent (in build file:/Users/dszeto/asf/incubator-predictionio/)
[error] error: Group field must be present in package: (main package)
[error] error: Package has no %description: predictionio-0.11.0_SNAPSHOT-1.noarch
[info] Building target platforms: noarch-apache-Linux
java.lang.RuntimeException: Unable to run rpmbuild, check output for details. Errorcode 1
	at scala.sys.package$.error(package.scala:27)
	at com.typesafe.sbt.packager.rpm.RpmHelper$$anonfun$buildPackage$1.apply(RpmHelper.scala:91)
	at com.typesafe.sbt.packager.rpm.RpmHelper$$anonfun$buildPackage$1.apply(RpmHelper.scala:68)
	at sbt.IO$.withTemporaryDirectory(IO.scala:344)
	at com.typesafe.sbt.packager.rpm.RpmHelper$.buildPackage(RpmHelper.scala:68)
	at com.typesafe.sbt.packager.rpm.RpmHelper$.buildRpm(RpmHelper.scala:20)
	at com.typesafe.sbt.packager.rpm.RpmPlugin$$anonfun$projectSettings$46.apply(RpmPlugin.scala:161)
	at com.typesafe.sbt.packager.rpm.RpmPlugin$$anonfun$projectSettings$46.apply(RpmPlugin.scala:157)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (assembly/rpm:packageBin) Unable to run rpmbuild, check output for details. Errorcode 1
[error] Total time: 0 s, completed Mar 28, 2017 10:25:38 AM
```

```
$rpmbuild --version
rpmbuild (RPM) 5.4.15
```

I am on macOS Sierra 10.12.3. Have you run into this?

**Pull request closed without merging**